scrypto-test/src/ledger_simulator/ledger_simulator.rs

```diff
@@ -1126,89 +1125,89 @@ impl<E: NativeVmExtension, D: TestDatabase> LedgerSimulator<E, D> {
         self.publish_package_with_owner(package_dir.as_ref(), owner_badge)
     }
 
-    pub fn execute_manifest<T>(
+    fn resolve_suggested_config(&self, manifest: &impl BuildableManifest) -> ExecutionConfig {
+        match manifest.suggested_execution_config_type() {
```

Comment:
`suggested_execution_config_type()` seems like being used only for tests.
So how about using below approach and ditch the `suggested_execution_config_type()` and `SuggestedExecutionConfigType` (Notarized variant is not set anywhere anyway)?
```
        let manifest = AnyTransactionManifest::from(manifest);
        match manifest {
            AnyTransactionManifest::V1(m)
            | AnyTransactionManifest::V2(m)
            | AnyTransactionManifest::SubintentV2(m) => ExecutionConfig::for_test_transaction(),
            AnyTransactionManifest::SystemV1(m) => {
                ExecutionConfig::for_system_transaction(NetworkDefinition::simulator())
            }
        }
        .with_kernel_trace(self.with_kernel_trace)
```

Reply:
It's an interesting idea... But I only created `AnyTransactionManifest` for the occasions where we can't have a concrete type to work at. It feels weird to me to map into it just to get an execution config and then map out of it in order to execute.

I guess I'd like to understand what you don't like about the current model? I guess that (a) it's not clear that it's used only for tests, and (b) introducing a new enum for this feels a bit over-the-top. I agree with both critiques (a) and (b).

Perhaps a method `default_test_execution_config(network_definition: NetworkDefinition)` might be more clearer / direct? :D. Will proceed with that, but happy to hear more thoughts from you on this! 👍

Reply:
Perhaps `fn default_test_execution_config_type(&self) -> DefaultTestExecutionConfigType;`?